homeassistant/components/yale_smart_alarm/binary_sensor.py

```diff
@@ -0,0 +1,59 @@
+"""Binary sensors for Yale Alarm."""
+from __future__ import annotations
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import COORDINATOR, DOMAIN, MANUFACTURER, MODEL
+from .coordinator import YaleDataUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Yale binary sensor entry."""
+
+    coordinator: YaleDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id][
+        COORDINATOR
+    ]
+
+    async_add_entities(
+        YaleBinarySensor(coordinator, data) for data in coordinator.data["door_windows"]
+    )
+
+
+class YaleBinarySensor(CoordinatorEntity, BinarySensorEntity):
+    """Representation of a Yale binary sensor."""
+
+    def __init__(self, coordinator: YaleDataUpdateCoordinator, data: dict) -> None:
+        """Initialize the Yale Lock Device."""
+        super().__init__(coordinator)
+        self._coordinator = coordinator
+        self._attr_name = data["name"]
+        self._attr_unique_id = data["address"]
+        self._attr_device_class = BinarySensorDeviceClass.DOOR
+        self._attr_device_info = DeviceInfo(
+            name=self._attr_name,
+            manufacturer=MANUFACTURER,
+            model=MODEL,
+            identifiers={(DOMAIN, data["address"])},
+            via_device=(DOMAIN, self._coordinator.entry.data[CONF_USERNAME]),
+        )
```

Comment:
This looks like the same as `lock.py`.  We should move it into a base entity class to make it more DRY.  Lots of examples:

```
% ls */entity.py
advantage_air/entity.py			harmony/entity.py			ozw/entity.py				steamist/entity.py
aseko_pool_live/entity.py		hassio/entity.py			plaato/entity.py			subaru/entity.py
august/entity.py			home_connect/entity.py			powerwall/entity.py			surepetcare/entity.py
balboa/entity.py			homematic/entity.py			rachio/entity.py			switchbot/entity.py
bond/entity.py				hunterdouglas_powerview/entity.py	rest/entity.py				tado/entity.py
bosch_shc/entity.py			ipp/entity.py				ring/entity.py				tplink/entity.py
broadlink/entity.py			isy994/entity.py			risco/entity.py				tractive/entity.py
device_automation/entity.py		juicenet/entity.py			rituals_perfume_genie/entity.py		unifiprotect/entity.py
devolo_home_network/entity.py		litterrobot/entity.py			roku/entity.py				uptimerobot/entity.py
directv/entity.py			lookin/entity.py			senseme/entity.py			wemo/entity.py
doorbird/entity.py			mobile_app/entity.py			shelly/entity.py			yeelight/entity.py
ezviz/entity.py				nanoleaf/entity.py			smarttub/entity.py			zha/entity.py
firmata/entity.py			nexia/entity.py				somfy/entity.py				zwave_js/entity.py
flo/entity.py				oncue/entity.py				sonarr/entity.py
flux_led/entity.py			opengarage/entity.py			sonos/entity.py
gree/entity.py				overkiz/entity.py			starline/entity.py
```

Reply:
Good suggestion. Coming up shortly and will make another PR to change lock to use same base entity